internal/charmstore/migrations.go

```diff
@@ -222,3 +226,35 @@ func migratePublishedEntities(db StoreDatabase) error {
 	}
 	return nil
 }
+
+// migrateCandidateBetaChannels populates base entity ACLs for the candidate
+// and beta channels.
+func migrateCandidateBetaChannels(db StoreDatabase) error {
+	baseEntities := db.BaseEntities()
+	iter := baseEntities.Find(bson.D{{
+		// Assume that, if a base entity does not have the "channelacls.beta"
+		// field, then the "channelacls.candidate" one is also missing and the
+		// document must be migrated.
+		"channelacls.beta", bson.D{{"$exists", false}},
+	}}).Select(map[string]int{"channelacls": 1}).Iter()
+
+	// For every resulting base entity populate "channelacls.beta" and
+	// "channelacls.candidate" with contents from "channelacls.unpublished".
```

Comment:
I think I agree that "edge" might be a better default, as "edge" is more restrictive than both "beta" and "candidate" as I understand it.